components/main-chef-wrapper/lib/dig.go

```diff
@@ -0,0 +1,44 @@
+package lib
+
```

Comment:
If this copied in from the original, we should attribute it with a comment, something like, based on the original version at https://... 

This file should also get a copyright 'tombstone'.

Reply:
I am not well aware about  copyright 'tombstone'., but some parts of snippet is modified and i have also added these statement now
``` 
//based on the original version at https://github.com/mnogu/go-dig
//Copyright (c) 2020 Muneyuki Noguchi
//Original Author: Muneyuki Noguchi <https://github.com/mnogu>
//Modified By: Pranay Singh <prsingh@progress.com>
```

Reply:
See below - we should have this in every new file, but the additional copyright and attribution 
is only for this one. 
```
// 
// Copyright (c) Chef Software, Inc.
// Copyright (c) 2020 Muneyuki Noguchi
// Based on the original by Muneyuki Noguchi at https://github.com/mnogu/go-dig
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

```